eval/src/vespa/eval/eval/test/value_compare.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "value_compare.h"
+#include <vespa/eval/eval/engine_or_factory.h>
+
+namespace vespalib::eval {
+
+bool operator==(const Value &lhs, const Value &rhs)
+{
+    auto engine = EngineOrFactory::get();
```

Comment:
we should not use EngineOrFactory::get in the eval library.

here we should keep and open mind about implementation, everywhere else, we should use EngineOrFactory::get to select the implementation.

This would not be useful for unit tests, since comparing values created by the SimpleTensorEngine would result in problems, since DefaultTensorEngine is the default tensor implementation.